### ISSUE TITLE
在vscode运行bash语句报错

### ISSUE THREAD

**Issue by xiatian815** [thuml/Koopa#10]
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.请问这是怎么回事，我在linux里面就可以正常执行代码

**Comment by lichenyu20** [comment]
1. 这个问题有可能是window上的线程机制引起的，为了避免这个问题，可以在其他平台上运行代码
2. 请您检查一下是否按照requirements.txt配置环境

**Comment by xiatian815** [comment]
你好，在vscode使用服务器节点运行bash仍然报错，这个是需要我设置参数吗。但是在本地虚拟机里面可以直接运行bash
$ bash ./scripts/ECL_script/Koopa.sh
usage: run.py [-h] --is_training IS_TRAINING --model_id MODEL_ID --model MODEL --data DATA [--root_path ROOT_PATH] [--data_path DATA_PATH] [--features FEATURES] [--target TARGET]
              [--freq FREQ] [--checkpoints CHECKPOINTS] [--seq_len SEQ_LEN] [--label_len LABEL_LEN] [--pred_len PRED_LEN] [--enc_in ENC_IN] [--dec_in DEC_IN] [--c_out C_OUT]
              [--dropout DROPOUT] [--embed EMBED] [--do_predict] [--num_workers NUM_WORKERS] [--itr ITR] [--train_epochs TRAIN_EPOCHS] [--batch_size BATCH_SIZE] [--patience PATIENCE]
              [--learning_rate LEARNING_RATE] [--des DES] [--loss LOSS] [--lradj LRADJ] [--use_amp] [--use_gpu USE_GPU] [--gpu GPU] [--use_multi_gpu] [--devices DEVICES] [--seed SEED]
              [--dynamic_dim DYNAMIC_DIM] [--hidden_dim HIDDEN_DIM] [--hidden_layers HIDDEN_LAYERS] [--seg_len SEG_LEN] [--num_blocks NUM_BLOCKS] [--alpha ALPHA] [--multistep]
run.py: error: the following arguments are required: --is_training, --model_id, --model, --data
./scripts/ECL_script/Koopa.sh: line 2: --is_training: command not found
./scripts/ECL_script/Koopa.sh: line 3: --root_path: command not found
./scripts/ECL_script/Koopa.sh: line 4: --data_path: command not found
./scripts/ECL_script/Koopa.sh: line 5: --model_id: command not found
./scripts/ECL_script/Koopa.sh: line 6: --model: command not found
./scripts/ECL_script/Koopa.sh: line 7: --data: command not found
./scripts/ECL_script/Koopa.sh: line 8: --features: command not found
./scripts/ECL_script/Koopa.sh: line 9: --seq_len: command not found
./scripts/ECL_script/Koopa.sh: line 10: --pred_len: command not found
./scripts/ECL_script/Koopa.sh: line 11: --seg_len: command not found
./scripts/ECL_script/Koopa.sh: line 12: --dynamic_dim: command not found
./scripts/ECL_script/Koopa.sh: line 13: --hidden_dim: command not found
./scripts/ECL_script/Koopa.sh: line 14: --hidden_layers: command not found
./scripts/ECL_script/Koopa.sh: line 15: --num_blocks: command not found
./scripts/ECL_script/Koopa.sh: line 16: --enc_in: command not found
./scripts/ECL_script/Koopa.sh: line 17: --dec_in: command not found
./scripts/ECL_script/Koopa.sh: line 18: --c_out: command not found
./scripts/ECL_script/Koopa.sh: line 19: --des: command not found
./scripts/ECL_script/Koopa.sh: line 20: --learning_rate: command not found
./scripts/ECL_script/Koopa.sh: line 21: --itr: command not found
./scripts/ECL_script/Koopa.sh: line 22: --gpu: command not found
./scripts/ECL_script/Koopa.sh: line 23: $'\r': command not found
usage: run.py [-h] --is_training IS_TRAINING --model_id MODEL_ID --model MODEL --data DATA [--root_path ROOT_PATH] [--data_path DATA_PATH] [--features FEATURES] [--target TARGET]
              [--freq FREQ] [--checkpoints CHECKPOINTS] [--seq_len SEQ_LEN] [--label_len LABEL_LEN] [--pred_len PRED_LEN] [--enc_in ENC_IN] [--dec_in DEC_IN] [--c_out C_OUT]
              [--dropout DROPOUT] [--embed EMBED] [--do_predict] [--num_workers NUM_WORKERS] [--itr ITR] [--train_epochs TRAIN_EPOCHS] [--batch_size BATCH_SIZE] [--patience PATIENCE]
              [--learning_rate LEARNING_RATE] [--des DES] [--loss LOSS] [--lradj LRADJ] [--use_amp] [--use_gpu USE_GPU] [--gpu GPU] [--use_multi_gpu] [--devices DEVICES] [--seed SEED]
              [--dynamic_dim DYNAMIC_DIM] [--hidden_dim HIDDEN_DIM] [--hidden_layers HIDDEN_LAYERS] [--seg_len SEG_LEN] [--num_blocks NUM_BLOCKS] [--alpha ALPHA] [--multistep]
run.py: error: the following arguments are required: --is_training, --model_id, --model, --data
./scripts/ECL_script/Koopa.sh: line 25: --is_training: command not found
./scripts/ECL_script/Koopa.sh: line 26: --root_path: command not found
./scripts/ECL_script/Koopa.sh: line 27: --data_path: command not found
./scripts/ECL_script/Koopa.sh: line 28: --model_id: command not found
./scripts/ECL_script/Koopa.sh: line 29: --model: command not found
./scripts/ECL_script/Koopa.sh: line 30: --data: command not found
./scripts/ECL_script/Koopa.sh: line 31: --features: command not found
./scripts/ECL_script/Koopa.sh: line 32: --seq_len: command not found
./scripts/ECL_script/Koopa.sh: line 33: --pred_len: command not found
./scripts/ECL_script/Koopa.sh: line 34: --seg_len: command not found
./scripts/ECL_script/Koopa.sh: line 35: --dynamic_dim: command not found
./scripts/ECL_script/Koopa.sh: line 36: --hidden_dim: command not found
./scripts/ECL_script/Koopa.sh: line 37: --hidden_layers: command not found
./scripts/ECL_script/Koopa.sh: line 38: --num_blocks: command not found
./scripts/ECL_script/Koopa.sh: line 39: --enc_in: command not found
./scripts/ECL_script/Koopa.sh: line 40: --dec_in: command not found
./scripts/ECL_script/Koopa.sh: line 41: --c_out: command not found
./scripts/ECL_script/Koopa.sh: line 42: --des: command not found
./scripts/ECL_script/Koopa.sh: line 43: --learning_rate: command not found
./scripts/ECL_script/Koopa.sh: line 44: --itr: command not found
./scripts/ECL_script/Koopa.sh: line 45: --gpu: command not found
./scripts/ECL_script/Koopa.sh: line 46: $'\r': command not found
usage: run.py [-h] --is_training IS_TRAINING --model_id MODEL_ID --model MODEL --data DATA [--root_path ROOT_PATH] [--data_path DATA_PATH] [--features FEATURES] [--target TARGET]
              [--freq FREQ] [--checkpoints CHECKPOINTS] [--seq_len SEQ_LEN] [--label_len LABEL_LEN] [--pred_len PRED_LEN] [--enc_in ENC_IN] [--dec_in DEC_IN] [--c_out C_OUT]
              [--dropout DROPOUT] [--embed EMBED] [--do_predict] [--num_workers NUM_WORKERS] [--itr ITR] [--train_epochs TRAIN_EPOCHS] [--batch_size BATCH_SIZE] [--patience PATIENCE]
              [--learning_rate LEARNING_RATE] [--des DES] [--loss LOSS] [--lradj LRADJ] [--use_amp] [--use_gpu USE_GPU] [--gpu GPU] [--use_multi_gpu] [--devices DEVICES] [--seed SEED]
              [--dynamic_dim DYNAMIC_DIM] [--hidden_dim HIDDEN_DIM] [--hidden_layers HIDDEN_LAYERS] [--seg_len SEG_LEN] [--num_blocks NUM_BLOCKS] [--alpha ALPHA] [--multistep]
run.py: error: the following arguments are required: --is_training, --model_id, --model, --data
./scripts/ECL_script/Koopa.sh: line 48: --is_training: command not found
./scripts/ECL_script/Koopa.sh: line 49: --root_path: command not found
./scripts/ECL_script/Koopa.sh: line 50: --data_path: command not found
./scripts/ECL_script/Koopa.sh: line 51: --model_id: command not found
./scripts/ECL_script/Koopa.sh: line 52: --model: command not found
./scripts/ECL_script/Koopa.sh: line 53: --data: command not found
./scripts/ECL_script/Koopa.sh: line 54: --features: command not found
./scripts/ECL_script/Koopa.sh: line 55: --seq_len: command not found
./scripts/ECL_script/Koopa.sh: line 56: --pred_len: command not found
./scripts/ECL_script/Koopa.sh: line 57: --seg_len: command not found
./scripts/ECL_script/Koopa.sh: line 58: --dynamic_dim: command not found
./scripts/ECL_script/Koopa.sh: line 59: --hidden_dim: command not found
./scripts/ECL_script/Koopa.sh: line 60: --hidden_layers: command not found
./scripts/ECL_script/Koopa.sh: line 61: --num_blocks: command not found
./scripts/ECL_script/Koopa.sh: line 62: --enc_in: command not found
./scripts/ECL_script/Koopa.sh: line 63: --dec_in: command not found
./scripts/ECL_script/Koopa.sh: line 64: --c_out: command not found
./scripts/ECL_script/Koopa.sh: line 65: --des: command not found
./scripts/ECL_script/Koopa.sh: line 66: --learning_rate: command not found
./scripts/ECL_script/Koopa.sh: line 67: --itr: command not found
./scripts/ECL_script/Koopa.sh: line 68: --gpu: command not found
./scripts/ECL_script/Koopa.sh: line 69: $'\r': command not found
usage: run.py [-h] --is_training IS_TRAINING --model_id MODEL_ID --model MODEL --data DATA [--root_path ROOT_PATH] [--data_path DATA_PATH] [--features FEATURES] [--target TARGET]
              [--freq FREQ] [--checkpoints CHECKPOINTS] [--seq_len SEQ_LEN] [--label_len LABEL_LEN] [--pred_len PRED_LEN] [--enc_in ENC_IN] [--dec_in DEC_IN] [--c_out C_OUT]
              [--dropout DROPOUT] [--embed EMBED] [--do_predict] [--num_workers NUM_WORKERS] [--itr ITR] [--train_epochs TRAIN_EPOCHS] [--batch_size BATCH_SIZE] [--patience PATIENCE]
              [--learning_rate LEARNING_RATE] [--des DES] [--loss LOSS] [--lradj LRADJ] [--use_amp] [--use_gpu USE_GPU] [--gpu GPU] [--use_multi_gpu] [--devices DEVICES] [--seed SEED]
              [--dynamic_dim DYNAMIC_DIM] [--hidden_dim HIDDEN_DIM] [--hidden_layers HIDDEN_LAYERS] [--seg_len SEG_LEN] [--num_blocks NUM_BLOCKS] [--alpha ALPHA] [--multistep]
run.py: error: the following arguments are required: --is_training, --model_id, --model, --data
./scripts/ECL_script/Koopa.sh: line 71: --is_training: command not found
./scripts/ECL_script/Koopa.sh: line 72: --root_path: command not found
./scripts/ECL_script/Koopa.sh: line 73: --data_path: command not found
./scripts/ECL_script/Koopa.sh: line 74: --model_id: command not found
./scripts/ECL_script/Koopa.sh: line 75: --model: command not found
./scripts/ECL_script/Koopa.sh: line 76: --data: command not found
./scripts/ECL_script/Koopa.sh: line 77: --features: command not found
./scripts/ECL_script/Koopa.sh: line 78: --seq_len: command not found
./scripts/ECL_script/Koopa.sh: line 79: --pred_len: command not found
./scripts/ECL_script/Koopa.sh: line 80: --seg_len: command not found
./scripts/ECL_script/Koopa.sh: line 81: --dynamic_dim: command not found
./scripts/ECL_script/Koopa.sh: line 82: --hidden_dim: command not found
./scripts/ECL_script/Koopa.sh: line 83: --hidden_layers: command not found
./scripts/ECL_script/Koopa.sh: line 84: --num_blocks: command not found
./scripts/ECL_script/Koopa.sh: line 85: --enc_in: command not found
./scripts/ECL_script/Koopa.sh: line 86: --dec_in: command not found
./scripts/ECL_script/Koopa.sh: line 87: --c_out: command not found
./scripts/ECL_script/Koopa.sh: line 88: --des: command not found
./scripts/ECL_script/Koopa.sh: line 89: --learning_rate: command not found
./scripts/ECL_script/Koopa.sh: line 90: --itr: command not found
./scripts/ECL_script/Koopa.sh: line 91: --gpu: command not found
